src/kaocha/specs.clj

```diff
@@ -1,27 +1,40 @@
 (ns kaocha.specs
   (:require [clojure.spec.alpha :as s]
             [clojure.test :as t]
-            [expound.alpha :as expound]))
+            [expound.alpha :as expound])
+  (:import (java.io FileNotFoundException)))
 
 (try
   (require 'clojure.test.check.generators)
-  (catch java.io.FileNotFoundException _))
-
-(def global-opts [:kaocha/reporter
-                  :kaocha/color?
-                  :kaocha/fail-fast?
-                  :kaocha/watch?
-                  :kaocha/plugins])
+  (catch FileNotFoundException _))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; global
 
+(s/def :kaocha/color? boolean?)
+
+(s/def :kaocha/fail-fast? boolean?)
+
+(s/def :kaocha/watch? boolean?)
+
 (s/def :kaocha/plugins (s/coll-of keyword?))
 
+(s/def :kaocha/reporter (s/or :fn      (s/fspec :args (s/cat :m map?))
+                              :symbol  symbol?
+                              :symbols (s/coll-of symbol? :kind vector?)))
+
+(s/def :kaocha/global-opts
+  (s/keys :opt [:kaocha/reporter
+                :kaocha/color?
+                :kaocha/fail-fast?
+                :kaocha/watch?
+                :kaocha/plugins]))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; config
 
-(s/def :kaocha/config (s/keys :req ~(conj global-opts :kaocha/tests)))
```

Comment:
That probably would make sense. When we read `tests.edn` there's normally a `#kaocha/v1` which does the processing/normalizing, so you should have a complete config map after that. But it would be useful to give people feedback on their config before processing, and their are cases where we get unprocessed or partial config maps, like when calling `kaocha.repl`.